modules/spring-config/src/main/java/org/mule/runtime/config/internal/DependencyGraphLifecycleObjectSorter.java

```diff
@@ -45,7 +44,7 @@
   private List<DefaultDirectedGraph<BeanWrapper, DefaultEdge>> dependencyGraphs;
   private DependencyGraphBeanDependencyResolver resolver;
   protected final Class<?>[] orderedLifecycleTypes;
-  private Map<String, Integer> lifecycleObjectNameOrderMap;
+  private List<String> lifecycleObjectList;
```

Comment:
This one was better off being a `Map` from component to priority (for tie-breaking). Having it internally as a list should be less performant because each look up is O(N) (and we probably need to do many of those).
The only scenario in which this could be better is if the Maps are very small (less than 5 elements) and the hash function of the keys is very expensive. In that case we can use a `SmallMap`.